src/openrct2/ride/Vehicle.cpp

```diff
@@ -7354,17 +7353,16 @@ static void vehicle_update_additional_animation(rct_vehicle* vehicle)
         case VEHICLE_ENTRY_ANIMATION_MULTI_DIM_COASTER: // loc_6D65E1
             if (vehicle->seat_rotation != vehicle->target_seat_rotation)
             {
-                if (vehicle->var_C8 <= 0xCCCC)
+                if (vehicle->additional_animation_var <= 0xCCCC)
                 {
-                    vehicle->var_C8 += 0x3333;
+                    vehicle->additional_animation_var += 0x3333;
                 }
                 else
                 {
-                    vehicle->var_C8 += 0x3333;
+                    vehicle->additional_animation_var = 0;
```

Comment:
Why this change?

Reply:
Previously this relied on rolling over. This just makes it explicit that it is resetting to 0.

Reply:
What if the previously value is not exactly 0xCCCC, but higher? It wouldn't be 0 then.

Reply:
Current behavior:

On the first cycle, it updates the frame when animationState is 0xFFFF (0x3333 + 0xCCCC). 0x3333 + 0xFFFF is 0x3332.
On the next cycle, it updates the frame when animationState is 0xFFFE (0x3333 + 0xCCCB), and decreases by one every future update.

After 0x3333 frame updates, animationState will finally reach zero. I was concerned that it might insert a 6th frame somewhere, but I think it won't do that.

Reply:
Even if it did insert that extra frame every 0x3333 cycles it would be silly to keep it like that.